InterfaceSwig/generated/tasmanian.f03

```diff
@@ -0,0 +1,3848 @@
+  procedure :: clearRefinement => swigf_TasmanianSparseGrid_clearRefinement
+  procedure :: mergeRefinement => swigf_TasmanianSparseGrid_mergeRefinement
+  procedure :: beginConstruction => swigf_TasmanianSparseGrid_beginConstruction
+  procedure :: isUsingConstruction => swigf_TasmanianSparseGrid_isUsingConstruction
+  procedure, private :: swigf_TasmanianSparseGrid_loadConstructedPoints__SWIG_1
+  procedure :: finishConstruction => swigf_TasmanianSparseGrid_finishConstruction
+  procedure, private :: swigf_TasmanianSparseGrid_setHierarchicalCoefficients__SWIG_1
+  procedure, private :: swigf_TasmanianSparseGrid_evaluateHierarchicalFunctions__SWIG_2
+  procedure, private :: swigf_TasmanianSparseGrid_integrateHierarchicalFu1JSVB3__SWIG_2
+  procedure, private :: swigf_TasmanianSparseGrid_printStats__SWIG_1
+  procedure, private :: swigf_TasmanianSparseGrid_enableAcceleration__SWIG_0
+  procedure, private :: swigf_TasmanianSparseGrid_enableAcceleration__SWIG_1
+  procedure, private :: swigf_TasmanianSparseGrid_enableAcceleration__SWIG_2
+  procedure, private :: swigf_TasmanianSparseGrid_enableAcceleration__SWIG_3
+  procedure :: favorSparseAcceleration => swigf_TasmanianSparseGrid_favorSparseAcceleration
+  procedure :: getAccelerationType => swigf_TasmanianSparseGrid_getAccelerationType
+  procedure, nopass :: isAccelerationAvailable => swigf_TasmanianSparseGrid_isAccelerationAvailable
+  procedure :: setGPUID => swigf_TasmanianSparseGrid_setGPUID
+  procedure :: getGPUID => swigf_TasmanianSparseGrid_getGPUID
+  procedure, nopass :: getNumGPUs => swigf_TasmanianSparseGrid_getNumGPUs
+  procedure, nopass :: getGPUMemory => swigf_TasmanianSparseGrid_getGPUMemory
+  procedure, nopass :: getGPUName => swigf_TasmanianSparseGrid_getGPUName
+  procedure, private :: swigf_TasmanianSparseGrid_removePointsByHierarchi1EK9TY__SWIG_0
+  procedure, private :: swigf_TasmanianSparseGrid_removePointsByHierarchi1EK9TY__SWIG_1
+  procedure, private :: swigf_TasmanianSparseGrid_removePointsByHierarchi1EK9TY__SWIG_2
+  procedure :: evaluateSparseHierarchicalFunctionsGetNZ => swigf_TasmanianSparseGrid_evaluateSparseHierarchicalFunct1CQM1I
+  procedure :: evaluateSparseHierarchicalFunctionsStatic => swigf_TasmanianSparseGrid_evaluateSparseHierarchicalFunct1HRJEE
+  procedure :: getPointsIndexes => swigf_TasmanianSparseGrid_getPointsIndexes
+  procedure :: getNeededIndexes => swigf_TasmanianSparseGrid_getNeededIndexes
+  procedure, private :: swigf_TasmanianSparseGrid_op_assign__
+  generic :: setDomainTransform => swigf_TasmanianSparseGrid_setDomainTransform__SWIG_1
+  generic :: getDomainTransform => swigf_TasmanianSparseGrid_getDomainTransform__SWIG_1
+  generic :: setAnisotropicRefinement => swigf_TasmanianSparseGrid_setAnisotropicRefinement__SWIG_1, &
+    swigf_TasmanianSparseGrid_setAnisotropicRefinement__SWIG_2
+  generic :: setSurplusRefinement => swigf_TasmanianSparseGrid_setSurplusRefinement__SWIG_1, &
```

Comment:
I got to the point where I can generate the files locally, but I am getting errors when I try to compile.

Gfortran claims this is ambiguous, probably because _SWIG_1 and _SWIG_7 have the same signature. On the C++ side, the difference is between an int and an enum.

Reply:
Aha, that makes sense. Is the C++ class overloaded between an int and enum? We'll have to instruct SWIG to `%ignore` one of the methods (passing in the full signature).

Reply:
We cannot `%ignore` since the methods do different things, the int and the enum are not equivalent. In other interfaces, where overloads are not well supported, I split the overload family into two sets.

Reply:
Well, we can `%rename` one set of methods. So if I'm unfolding the overloads and default arguments correctly (and ignoring the additional overloads with `std::vector`, we have:
```
void setSurplusRefinement(double, int);
void setSurplusRefinement(double, int, const int *);
void setSurplusRefinement(double, TypeRefinement);
void setSurplusRefinement(double, TypeRefinement, int);
void setSurplusRefinement(double, TypeRefinement, int, const int *);
void setSurplusRefinement(double, TypeRefinement, int, const int *, const double *);
```

I think you can resolve this with:
```swig
%rename(setSurplusRefinementOutput) void TasGrid::TasmanianSparseGrid::setSurplusRefinement(double, int);
%rename(setSurplusRefinementOutput) void TasGrid::TasmanianSparseGrid::setSurplusRefinement(double, int, const int *);
```

Reply:
To avoid confusion, lets do the following:

Without an enum, the methods refer to `Global` version of the surplus refinement:
```
%rename(setGlobalSurplusRefinement) void TasGrid::TasmanianSparseGrid::setSurplusRefinement(double, int);
%rename(setGlobalSurplusRefinement) void TasGrid::TasmanianSparseGrid::setSurplusRefinement(double, int, const int *);
```

And for the enum refers specifically to a Local sub-strategy for surplus refinement:
```
%rename(setLocalSurplusRefinement) void TasGrid::TasmanianSparseGrid::setSurplusRefinement(double, TypeRefinement);
%rename(setLocalSurplusRefinement) void TasGrid::TasmanianSparseGrid::setSurplusRefinement(double, TypeRefinement, int);
%rename(setLocalSurplusRefinement) void TasGrid::TasmanianSparseGrid::setSurplusRefinement(double, TypeRefinement, int, const int *);
%rename(setLocalSurplusRefinement) void TasGrid::TasmanianSparseGrid::setSurplusRefinement(double, TypeRefinement, int, const int *, const double *);
```

Reply:
Might be worth changing the C++ interface itself in that case, actually... I did not immediately see that set of functions was global and the other local.

Reply:
You might be right, I have to think some more.

The enum criteria specifies one of the refinement variations, but global grids (defined by previous use of makeGlobalGrid()) are very rigid and do not allow for variations. 

The rest of the inputs are the same and this is one of the few cases where the underlying grid type matters for interface purposes.

Reply:
@sethrj You had a point on the C++ interface, I will make a change.

For now, we can just ignore the overloads without enums. I will do that, I need to get my feet wet a bit with the swift code anyway. I'm also in the middle of making other changes, so don't change anything on your side.

Reply:
Great! Let me know if you need me to point you in the right direction on anything SWIG-related.